prow/gangway/gangway.go

```diff
@@ -135,13 +147,36 @@ func (gw *Gangway) GetJobExecution(ctx context.Context, gjer *GetJobExecutionReq
 		jobStatus = JobExecutionStatus_JOB_EXECUTION_STATUS_UNSPECIFIED
 
 	}
+	return jobStatus
+}
 
-	jobExec := &JobExecution{
-		Id:        prowJobCR.Name,
-		JobStatus: jobStatus,
+func (gw *Gangway) BulkJobStatusChange(ctx context.Context, bjscr *BulkJobStatusChangeRequest) (*JobExecutions, error) {
+	if err := bjscr.Validate(); err != nil {
+		return nil, err
 	}
 
-	return jobExec, nil
+	pjStateString := bjscr.GetJobStatusChange().GetCurrent().String()
+	selector := fields.SelectorFromSet(map[string]string{"status.state": pjStateString})
+	pjList, err := gw.ProwJobClient.List(context.TODO(), metav1.ListOptions{FieldSelector: selector.String()})
+	if err != nil {
+		return nil, err
+	}
+
+	jobExecutions := &JobExecutions{}
+	for _, pj := range pjList.Items {
+		pj.Status.State = prowcrd.ProwJobState(bjscr.GetJobStatusChange().GetDesired().String())
+		newPj, err := gw.ProwJobClient.UpdateStatus(context.TODO(), &pj, metav1.UpdateOptions{})
```

Comment:
Be careful with this when testing. You are actually changing the status, which is desired, but make sure you don't use this on prod data until it is ready

Reply:
yes, thanks for the reminder, I am only trying to make it work in the `ci-staging` ns